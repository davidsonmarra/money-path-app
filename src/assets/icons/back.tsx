import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const BackIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-back">
    <Path d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z" />
  </Svg>
);

export default BackIcon;
