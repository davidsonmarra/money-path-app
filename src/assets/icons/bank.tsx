import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const BankIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-bank">
    <Path d="M200-280v-280h80v280zm240 0v-280h80v280zM80-120v-80h800v80zm600-160v-280h80v280zM80-640v-80l400-200 400 200v80zm178-80h444zm0 0h444L480-830z" />
  </Svg>
);

export default BankIcon;
