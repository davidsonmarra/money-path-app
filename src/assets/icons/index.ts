export {default as AppleIcon} from 'src/assets/icons/apple';
export {default as BackIcon} from 'src/assets/icons/back';
export {default as BancoDoBrasilIcon} from 'src/assets/icons/banco-do-brasil';
export {default as BankIcon} from 'src/assets/icons/bank';
export {default as CheckIcon} from 'src/assets/icons/check';
export {default as ChevronRightIcon} from 'src/assets/icons/chevron-right';
export {default as CloseIcon} from 'src/assets/icons/close';
export {default as ErrorIcon} from 'src/assets/icons/error';
export {default as GoogleIcon} from 'src/assets/icons/google';
export {default as InterIcon} from 'src/assets/icons/inter';
export {default as ItauIcon} from 'src/assets/icons/itau';
export {default as MoneyPathIcon} from 'src/assets/icons/money-path';
export {default as NubankIcon} from 'src/assets/icons/nubank';
export {default as SantanderIcon} from 'src/assets/icons/santander';
export {default as WalletIcon} from 'src/assets/icons/wallet';
export {default as WarningIcon} from 'src/assets/icons/warning';
