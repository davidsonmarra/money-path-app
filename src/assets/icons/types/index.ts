export enum IconType {
  apple = 'apple',
  back = 'back',
  bancoDoBrasil = 'bancoDoBrasil',
  bank = 'bank',
  check = 'check',
  chevronRight = 'chevronRight',
  close = 'close',
  error = 'error',
  food = 'food',
  games = 'games',
  google = 'google',
  health = 'health',
  help = 'help',
  house = 'house',
  inter = 'inter',
  itau = 'itau',
  moneyPath = 'moneyPath',
  money = 'money',
  nubank = 'nubank',
  santander = 'santander',
  transport = 'transport',
  wallet = 'wallet',
  warning = 'warning',
}
