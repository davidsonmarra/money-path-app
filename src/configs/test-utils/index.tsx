import {render} from '@testing-library/react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import AddWalletFormContext from 'src/features/add-wallet/context/add-wallet-form';

interface IAllThemeProviders {
  children: React.ReactElement;
}

const AllTheProviders = ({children}: IAllThemeProviders) => (
  <AddWalletFormContext>
    <SafeAreaView>{children}</SafeAreaView>
  </AddWalletFormContext>
);

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export const renderWithCustomProviders = (component: React.ReactElement) => {
  return {
    render: render(<SafeAreaView>{component}</SafeAreaView>),
  };
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
