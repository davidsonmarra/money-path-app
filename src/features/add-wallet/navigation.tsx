import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import FeedbackScreen from 'src/features/add-wallet/screens/feedback';
import InitialValueScreen from 'src/features/add-wallet/screens/initial-value';
import NameScreen from 'src/features/add-wallet/screens/name';
import SelectColorScreen from 'src/features/add-wallet/screens/select-color';
import SelectInstitutionScreen from 'src/features/add-wallet/screens/select-institution';
import AddWalletFormContext from 'src/features/add-wallet/context/add-wallet-form';

export type AddWalletStackParamList = {
  Feedback: undefined;
  InitialValue: undefined;
  Name: undefined;
  SelectColor: undefined;
  SelectInstitution: undefined;
};

const { Navigator, Screen } = createStackNavigator<AddWalletStackParamList>();

const AddWalletStack = () => {
  return (
    <AddWalletFormContext>
      <Navigator
        initialRouteName="SelectInstitution"
        screenOptions={{
          headerShown: false,
        }}
      >
        <Screen name="Feedback" component={FeedbackScreen} />
        <Screen name="InitialValue" component={InitialValueScreen} />
        <Screen name="Name" component={NameScreen} />
        <Screen name="SelectColor" component={SelectColorScreen} />
        <Screen name="SelectInstitution" component={SelectInstitutionScreen} />
      </Navigator>
    </AddWalletFormContext>
  );
};

export default AddWalletStack;
