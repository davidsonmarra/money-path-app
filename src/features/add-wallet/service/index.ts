export {default as createWallet} from 'src/features/add-wallet/service/create-wallet';
