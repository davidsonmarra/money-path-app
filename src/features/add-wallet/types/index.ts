import {IconType} from 'src/assets/icons/types';

export interface WalletProps {
  name: string;
  icon: IconType;
  color?: string;
  backgroundColor?: string;
  type: 'personal' | 'bank';
  amount?: number;
}

export type SelectWalletSection = {
  title: string;
  data: WalletProps[];
};
