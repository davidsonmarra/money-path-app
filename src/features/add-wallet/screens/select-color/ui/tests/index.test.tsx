import React from 'react';
import {IconType} from 'src/assets/icons/types';
import {fireEvent, render} from 'src/configs/test-utils';
import SelectColorContainer, {
  Props,
} from 'src/features/add-wallet/screens/select-color/ui';
import {InstitutionProps} from 'src/features/add-wallet/types';

let mockInstitution: InstitutionProps = {
  backgroundColor: '#2A5C99',
  color: '#EAC43D',
  icon: IconType.wallet,
  name: 'Carteira',
  type: 'personal',
};
const mockOnBackPress = jest.fn();
const mockSelectColor = jest
  .fn()
  .mockImplementation((institution: InstitutionProps) => {
    mockInstitution = institution;
  });
const mockOnConfirm = jest.fn();
const containerInstance = ({institution = mockInstitution}: Partial<Props>) =>
  render(
    <SelectColorContainer
      institution={institution}
      onBack={mockOnBackPress}
      selectColor={mockSelectColor}
      onConfirm={mockOnConfirm}
    />,
  );

describe('SelectColorContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const container = containerInstance({}).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should call onBackPress', () => {
    const container = containerInstance({});
    container.getByTestId('btn-left-icon').props.onClick();
    expect(mockOnBackPress).toHaveBeenCalledTimes(1);
  });

  it('should call selectInstitution for icon color', async () => {
    const container = containerInstance({});
    const colorButton = container.getByTestId('btn-color-#EE82EE');
    fireEvent.press(colorButton);
    expect(mockSelectColor).toHaveBeenCalledWith(mockInstitution);
  });
});
