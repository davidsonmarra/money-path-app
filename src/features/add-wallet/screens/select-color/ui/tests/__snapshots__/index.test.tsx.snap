// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectColorContainer should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2A5C99",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 12,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="btn-left-icon"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        fill="#FFF"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        testID="icon-back"
        vbHeight={960}
        vbWidth={960}
        width={24}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        [
          {
            "color": "#FFF",
            "fontSize": 20,
            "fontWeight": "bold",
          },
          {
            "color": "#FFF",
          },
        ]
      }
    >
      Personalize a Cor
    </Text>
    <View />
  </View>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "off",
        "right": "off",
        "top": "off",
      }
    }
    style={
      {
        "backgroundColor": "#232E38",
        "flex": 1,
        "justifyContent": "space-between",
        "paddingHorizontal": 24,
        "paddingVertical": 16,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#2A5C99",
            "borderRadius": 80,
            "height": 160,
            "justifyContent": "center",
            "marginBottom": 32,
            "marginHorizontal": "auto",
            "width": 160,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={100}
          bbWidth={100}
          fill="#EAC43D"
          focusable={false}
          height={100}
          meetOrSlice={0}
          minX={0}
          minY={-960}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 100,
                "width": 100,
              },
            ]
          }
          testID="icon-wallet"
          vbHeight={960}
          vbWidth={960}
          width={100}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4293575741,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M240-160q-66 0-113-47T80-320v-320q0-66 47-113t113-47h480q66 0 113 47t47 113v320q0 66-47 113t-113 47zm0-480h480q22 0 42 5t38 16v-21q0-33-23.5-56.5T720-720H240q-33 0-56.5 23.5T160-640v21q18-11 38-16t42-5m-74 130 445 108q9 2 18 0t17-8l139-116q-11-15-28-24.5t-37-9.5H240q-26 0-45.5 13.5T166-510"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        style={
          [
            {
              "color": "#D9D9D9",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Escolha uma cor para o Ã­cone:
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 32,
            "marginVertical": 8,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="btn-color-#FF0000"
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#FF0000",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="btn-color-#FFA500"
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#FFA500",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="btn-color-#EAC43D"
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#EAC43D",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="btn-color-#2A5C99"
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#2A5C99",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="btn-color-#008000"
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#008000",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="btn-color-#4B0082"
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#4B0082",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="btn-color-#EE82EE"
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#EE82EE",
                },
                false,
              ]
            }
          />
        </View>
      </View>
      <Text
        style={
          [
            {
              "color": "#D9D9D9",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Escolha uma cor para o fundo:
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 32,
            "marginVertical": 8,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#FF0000",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#FFA500",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#EAC43D",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#2A5C99",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#008000",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#4B0082",
                },
                false,
              ]
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 20,
                  "height": 40,
                  "margin": 4,
                  "opacity": 0.3,
                  "width": 40,
                },
                {
                  "backgroundColor": "#EE82EE",
                },
                false,
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#EAC43D",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
    >
      <View />
      <Text
        style={
          [
            {
              "color": "#FFF",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Continuar
      </Text>
      <View />
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaView>
`;
